evm/utils/headers.py

```diff
@@ -65,3 +69,29 @@ def compute_gas_limit(parent_header, gas_limit_floor):
         return parent_header.gas_limit + decay
     else:
         return gas_limit
+
+
+def generate_header_from_parent_header(
+        compute_difficulty,
+        parent_header,
+        timestamp,
+        coinbase=b'\x35' * 20,
+        extra_data=b''):
+    """
+    Generate BlockHeader from state_root and parent_header
+    """
+    header = BlockHeader(
+        difficulty=compute_difficulty(parent_header, timestamp),
+        block_number=(parent_header.block_number + 1),
+        gas_limit=compute_gas_limit(
+            parent_header,
+            gas_limit_floor=GENESIS_GAS_LIMIT,
+        ),
+        timestamp=max(timestamp, parent_header.timestamp + 1),
```

Comment:
I think that rather than using `max` here this should throw an exception if `timestamp <= parent_header.timestamp`.  It feels wrong to override the `timestamp` argument in this way.

Reply:
Yes, you're right, it was supposed to be checking current time (`time.time()`) and `parent_header.timestamp + 1` if `timestamp is None`. I'll update it. Thank you.